src/Hamiltonians/reduced_density_matrix.jl

```diff
@@ -121,3 +121,84 @@ function get_offdiagonal(
     address, value = excitation(addr, dst, src)
     return address, value
 end
+
+"""
+    ReducedDensityMatrix(addr::SingleComponentFockAddress; n = 1) <: AbstractOperator
+
+Represent the n-particle reduced density matrix:
+
+```math
+ρ̂^{(n)}_{j_1,...,j_1,k_1,...,k_n} =  \\prod_{i}^{n} â^†_{j_i} \\prod_{l}^{n} â_{k_{n+1-l}}
+```
+
+Where `j_i` and `k_i` (all `<: Int`) specify the single particle sites on a lattice.
+Additionally, the indices run in the following manners:
+
+```math
+j_n> ... > j_{i+1} > j_{i} > ... > j_1 and k_n> ... > k_{i+1} > k_{i} > ... > k_1
+```
+
+# See also
+
+* [`single_particle_density`](@ref)
+* [`SingleParticleDensity`](@ref)
+* [`SingleParticleExcitation`](@ref)
+* [`TwoParticleExcitation`](@ref)
+"""
+struct ReducedDensityMatrix <: AbstractOperator{Matrix{Float64}}
+    M::Int
+    n::Int
+end
+function ReducedDensityMatrix(addr::SingleComponentFockAddress; n = 1)
+    M = num_modes(addr)
+    return ReducedDensityMatrix(M,n)
+end
+function Base.show(io::IO, op::ReducedDensityMatrix)
+    print(io, "ReducedDensityMatrix(num_modes = $(op.M), n=$(op.n))")
+end
```

Comment:
The `show` function should print out a constructor for the type, so something that can be pasted into the REPL to re-construct the type. This one doesn't.